ibis/expr/operations.py

```diff
@@ -2450,6 +2450,11 @@ def output_type(self):
         return dt.timestamp.scalar_type()
 
 
+class Random(Constant):
```

Comment:
this could be RandomScalar. you could eventually expand this to a column op as well.

Reply:
@jrbourbeau do you still have any interesting on moving this forward? let me know if you need any help! thanks for having been working on that.

Reply:
Apologies for the delayed response. I've made updates in response to @jreback's initial feedback, please let me know if there's anything else you'd like me to include in this PR